# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/D3D11ShareHandleImage.h
# Commit: 6149f5b15c5e
# Full Hash: 6149f5b15c5e278ef15ea36fa896fa8c146ebdca
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-14 08:32:45
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 29fd3a6a7e27 -r 6149f5b15c5e gfx/layers/D3D11ShareHandleImage.h
--- a/gfx/layers/D3D11ShareHandleImage.h	Wed Jan 13 20:14:54 2021 +0000
+++ b/gfx/layers/D3D11ShareHandleImage.h	Wed Jan 13 21:49:40 2021 +0000
@@ -64,6 +64,8 @@
   TextureClient* GetTextureClient(KnowsCompositor* aKnowsCompositor) override;
   gfx::IntRect GetPictureRect() const override { return mPictureRect; }
 
+  Maybe<SurfaceDescriptor> GetDesc() override { return GetDescFromTexClient(); }
+
   ID3D11Texture2D* GetTexture() const;
 
   gfx::YUVColorSpace GetYUVColorSpace() const { return mYUVColorSpace; }